pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -120,8 +136,12 @@ func newCmdConfigSetAuthInfo(out io.Writer, options *createAuthInfoOptions) *cob
 	cmd.Flags().Var(&options.clientKey, clientcmd.FlagKeyFile, "path to "+clientcmd.FlagKeyFile+" file for the user entry in kubeconfig")
 	cmd.MarkFlagFilename(clientcmd.FlagKeyFile)
 	cmd.Flags().Var(&options.token, clientcmd.FlagBearerToken, clientcmd.FlagBearerToken+" for the user entry in kubeconfig")
+	cmd.Flags().BoolVar(&options.promptCredentials, "prompt-credentials", false, "prompt user for username and password")
```

Comment:
I mentioned both ideas but I don't really think we need *both* `--prompt-credentials` and `--prompt-username` + `prompt-password`. I'd suggest removing `--prompt-credentials` and keep the latter.